### ISSUE TITLE
Setup failed for custom integration 'pirateweather': Unable to import component after upgrading to Home Assistant 2024.10.0

### ISSUE THREAD

**Issue by dbruynb** [Pirate-Weather/pirate-weather-ha#321]
### Describe the bug

After Home Assistant upgrade I got

```
Setup failed for custom integration 'pirateweather': Unable to import component: cannot import name 'Forecast' from 'forecastio.models' (/usr/local/lib/python3.12/site-packages/forecastio/models.py)
```

Troubleshooting steps

1. Removed the integration
2. Rebooted Home Assistant
3. Adding it back again I get this error:
```
Error
Config flow could not be loaded: {"message":"Invalid handler specified"}
```

### Expected behavior

Expecting integration to work

### Actual behavior

See error logs:  

```
Logger: homeassistant.setup
Source: setup.py:334
First occurred: 8:16:26 AM (1 occurrences)
Last logged: 8:16:26 AM

Setup failed for custom integration 'pirateweather': Unable to import component: cannot import name 'Forecast' from 'forecastio.models' (/usr/local/lib/python3.12/site-packages/forecastio/models.py)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1007, in async_get_component
    comp = await self.hass.async_add_import_executor_job(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1067, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/pirateweather/__init__.py", line 34, in <module>
    from .weather_update_coordinator import WeatherUpdateCoordinator
  File "/config/custom_components/pirateweather/weather_update_coordinator.py", line 9, in <module>
    from forecastio.models import Forecast
ImportError: cannot import name 'Forecast' from 'forecastio.models' (/usr/local/lib/python3.12/site-packages/forecastio/models.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 334, in _async_setup_component
    component = await integration.async_get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1027, in async_get_component
    self._component_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1019, in async_get_component
    comp = self._get_component()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1067, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/pirateweather/__init__.py", line 34, in <module>
    from .weather_update_coordinator import WeatherUpdateCoordinator
  File "/config/custom_components/pirateweather/weather_update_coordinator.py", line 9, in <module>
    from forecastio.models import Forecast
ImportError: cannot import name 'Forecast' from 'forecastio.models' (/usr/local/lib/python3.12/site-packages/forecastio/models.py)
```

```
Logger: homeassistant.util.loop
Source: util/loop.py:77
First occurred: 8:16:26 AM (1 occurrences)
Last logged: 8:16:26 AM

Detected blocking call to import_module with args ('custom_components.pirateweather',) in /usr/src/homeassistant/homeassistant/loader.py, line 1067: ComponentProtocol, importlib.import_module(self.pkg_path) inside the event loop; This is causing stability issues. Please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#import_module Traceback (most recent call last): File "<frozen runpy>", line 198, in _run_module_as_main File "<frozen runpy>", line 88, in _run_code File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module> sys.exit(main()) File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main exit_code = runner.run(runtime_conf) File "/usr/src/homeassistant/homeassistant/runner.py", line 189, in run return loop.run_until_complete(setup_and_run_hass(runtime_config)) File "/usr/local/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete self.run_forever() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever self._run_once() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1990, in _run_once handle._run() File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run self._context.run(self._callback, *self._args) File "/usr/src/homeassistant/homeassistant/setup.py", line 165, in async_setup_component result = await _async_setup_component(hass, domain, config) File "/usr/src/homeassistant/homeassistant/setup.py", line 334, in _async_setup_component component = await integration.async_get_component() File "/usr/src/homeassistant/homeassistant/loader.py", line 1019, in async_get_component comp = self._get_component() File "/usr/src/homeassistant/homeassistant/loader.py", line 1067, in _get_component ComponentProtocol, importlib.import_module(self.pkg_path)
```

### Home Assistant version

2024.10.0

### Integration version

1.5.9

### Troubleshooting steps

- [X] I have updated my Home Assistant installation to the latest version.
- [X] I have updated the Pirate Weather Integration to the latest version.
- [X] I have gone through the [documentation](https://pirateweather.net/en/latest/ha/) before opening this issue.
- [X] I have searched this repository and [API Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have restarted my Home Assistant installation.
- [X] I have queried the API in my browser to confirm the issue is not with the API.
- [X] I have not included my API key anywhere in my issue.
- [X] I have written an informative title.

**Comment by cloneofghosts** [comment]
I updated my HA install last night and the devcontainer I use for testing this morning to 2024.10.0 and I've encountered no issues with the integration. I'll ping @alexander0042 to look at this to see if he can figure anything out.

The plan is to eventually stop using the forecastio library and move over to this one https://github.com/bakobako/pirate-weather-python (see the [future branch](https://github.com/Pirate-Weather/pirate-weather-ha/tree/future)) but it hasn't been worked on in a while.

**Comment by alexander0042** [comment]
Thanks for flagging this! I'll upgrade my instance and check it out, since it would be a big issue if the forcastio library has been depreciated somehow?

@dbruynb, can you add some details on what platform you're running HA on? Is it 32-64 bit, ARM or x86?

**Comment by dbruynb** [comment]
> Thanks for flagging this! I'll upgrade my instance and check it out, since it would be a big issue if the forcastio library has been depreciated somehow?
> 
> 
> 
> @dbruynb, can you add some details on what platform you're running HA on? Is it 32-64 bit, ARM or x86?

Im running it on a Home Assistant Blue. So that would be 64bit ARM

**Comment by cloneofghosts** [comment]
I wonder if this issue is only affecting ARM platforms as on my HAOS install it works fine. There was an issue created for a different issue and they were running the integration on 2024.10.0 without any issues.

Either way it's probably a good idea to start moving away from the forecastio library and use the new PW Python library instead as forecastio will probably break in a future python version.

**Comment by alexander0042** [comment]
Just finished updating my instance (Raspberry Pi 4) running HAOS, so similar platform, but no apparent issues on my end, so pretty weird. Maybe try reinstalling via HACS?

We really are barely using the old Forcastio library at this point, so really should do this ASAP. That linked repo is a great option, but since all we're using is their models, we could even just add a "models.py" file to our repo from the [original source](https://github.com/ZeevG/python-forecast.io/blob/17bc91b6672b651db013adfae9d4584db56ef49a/forecastio/models.py), and I think drop that dependency entirely! Since all this integration does is make a simple curl request to the endpoint and transform the json into something easier to work with, I don't see the need to make it any more complex.

**Comment by dbruynb** [comment]
Removing the integration from HACS and reinstalling also does not work, however, my issue is now resolved after upgrading to Home Assistant 2024.10.1
